Code/src/test/java/nl/utwente/viskell/ui/ToplevelPaneTest.java

```diff
@@ -0,0 +1,202 @@
+        toplevelPane.fromBundle(ImmutableMap.of()); // no crash
+    }
+
+    @Test
+    public void nullBlocksBundleFromBundleTest() {
+        // Setup
+        Map<String, Object> blocksBundle = new HashMap<>();
+        blocksBundle.put(ToplevelPane.BLOCKS_SERIALIZED_NAME, null);
+        // no connections bundle
+
+        // UUT
+        ToplevelPane toplevelPane = new ToplevelPane(mockGhci);
+
+        // Test
+        toplevelPane.fromBundle(blocksBundle); // no crash
+    }
+
+    @Test
+    public void nullConnectionsBundleFromBundleTest() {
+        // Setup
+        Map<String, Object> connectionsBundle = new HashMap<>();
+        connectionsBundle.put(ToplevelPane.CONNECTIONS_SERIALIZED_NAME, null);
+        // no blocks bundle
+
+        // UUT
+        ToplevelPane toplevelPane = new ToplevelPane(mockGhci);
+
+        // Test
+        toplevelPane.fromBundle(connectionsBundle); // no crash
+    }
+
+    @Test
+    public void emptyBlocksListFromBundleTest() {
+        // Setup
+        Map<String, Object> blocksBundle = new HashMap<>();
+        @SuppressWarnings("MismatchedQueryAndUpdateOfCollection")
+        List<Map<String, Object>> blocks = new ArrayList<>();
+        blocksBundle.put(ToplevelPane.BLOCKS_SERIALIZED_NAME, blocks);
+
+        // UUT
+        ToplevelPane toplevelPane = new ToplevelPane(mockGhci);
+
+        // Test
+        toplevelPane.fromBundle(blocksBundle); // no crash
+    }
+
+    @Test
+    public void emptyConnectionsListFromBundleTest() {
+        // Setup
+        Map<String, Object> connectionsBundle = new HashMap<>();
+        @SuppressWarnings("MismatchedQueryAndUpdateOfCollection")
+        List<Map<String, Object>> connections = new ArrayList<>();
+        connectionsBundle.put(ToplevelPane.CONNECTIONS_SERIALIZED_NAME, connections);
+
+        // UUT
+        ToplevelPane toplevelPane = new ToplevelPane(mockGhci);
+
+        // Test
+        toplevelPane.fromBundle(connectionsBundle); // no crash
+    }
+
+    /*
+    Couldn't get this test to work. To try it you should change the line at the start of the test class to be:
+    @PrepareForTest({ToplevelPane.class, Pane.class, Block.class})
+
+    I need to @PrepareForTest ToplevelPane.class to get whenNew() to work -
+    but it seems to fail due to a problem related to javassist and java8...
+
+    Error currently reported is:
+        "java.lang.IllegalStateException: Failed to transform class with name nl.utwente.viskell.ui.ToplevelPane.
+         Reason: javassist.bytecode"
+
+
+    Spent about a day in junit/powermock/javassist dependency hell trying many different versions
+    of junit and powermock that produce different failures I couldn't solve either....such as
+            "java.lang.VerifyError: Inconsistent stackmap frames at branch target 1135"
+            (that one seems to be Java 7 related and can be avoided passing "-noverify" to JVM when running tests,
+            but that seems a hack)
+
+    ....then I gave up...
```

Comment:
I think that's because the mocking transformation of the file (byte code manipulation) fails...